From ca8779363fd30a1b8fd80ce4ebacc4741b041c76 Mon Sep 17 00:00:00 2001
From: BogDan Vatra <bogdan@kdab.com>
Date: Tue, 20 Mar 2018 10:36:43 +0200
Subject: [PATCH] Android: Fix crash

Android doesn't like nor use RTLD_NODELETE

Tasnk-number: QTBUG-64654
Change-Id: I2d884bbf22a681cca592942eba84ba97327ba974
Reviewed-by: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@qt.io>
---
 src/corelib/plugin/qlibrary_unix.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git x/qtbase/src/corelib/plugin/qlibrary_unix.cpp y/qtbase/src/corelib/plugin/qlibrary_unix.cpp
index 23b9ad6434..296af9da7a 100644
--- x/qtbase/src/corelib/plugin/qlibrary_unix.cpp
+++ y/qtbase/src/corelib/plugin/qlibrary_unix.cpp
@@ -155,7 +155,7 @@ bool QLibraryPrivate::load_sys()
     // Do not unload the library during dlclose(). Consequently, the
     // library's specific static variables are not reinitialized if the
     // library is reloaded with dlopen() at a later time.
-#ifdef RTLD_NODELETE
+#if defined(RTLD_NODELETE) && !defined(Q_OS_ANDROID)
     if (loadHints & QLibrary::PreventUnloadHint) {
         dlFlags |= RTLD_NODELETE;
     }
-- 
2.19.1

